// ngx-md/markdown/markdown.service.js
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs/Observable";
import "rxjs/add/observable/of";
import "rxjs/add/operator/map";
import "rxjs/add/operator/catch";
import * as markedNs from "marked";
var MarkdownService = (function () {
    function MarkdownService(_http) {
        this._http = _http;
        this._renderer = new markedNs.Renderer();
        this.extendRenderer();
        this.setMarkedOptions({});
    }
    //get the content from remote resource
    //get the content from remote resource
    MarkdownService.prototype.getContent = 
    //get the content from remote resource
    function (path) {
        return this._http
            .get(path, { responseType: "text" })
            .map(this.extractData)
            .catch(this.handleError);
    };
    Object.defineProperty(MarkdownService.prototype, "renderer", {
        get: function () {
            return this._renderer;
        },
        enumerable: true,
        configurable: true
    });
    // handle data
    // handle data
    MarkdownService.prototype.extractData = 
    // handle data
    function (res) {
        return res || "";
    };
    MarkdownService.prototype.setMarkedOptions = function (options) {
        options = Object.assign({
            gfm: true,
            tables: true,
            breaks: false,
            pedantic: false,
            sanitize: true,
            smartLists: true,
            smartypants: false
        }, options);
        options.renderer = this._renderer;
        markedNs.setOptions(options);
    };
    // comple markdown to html
    // comple markdown to html
    MarkdownService.prototype.compile = 
    // comple markdown to html
    function (data) {
        return markedNs(data);
    };
    //handle error
    //handle error
    MarkdownService.prototype.handleError = 
    //handle error
    function (error) {
        var errMsg;
        if (error instanceof fetch) {
            var body = error.json() || "";
            var err = body.error || JSON.stringify(body);
            errMsg = error.status + " - " + (error.statusText || "") + " " + err;
        }
        else {
            errMsg = error.message ? error.message : error.toString();
        }
        return Observable.throw(errMsg);
    };
    // extend marked render to support todo checkbox
    // extend marked render to support todo checkbox
    MarkdownService.prototype.extendRenderer = 
    // extend marked render to support todo checkbox
    function () {
        this._renderer.listitem = function (text) {
            if (/^\s*\[[x ]\]\s*/.test(text)) {
                text = text
                    .replace(/^\s*\[ \]\s*/, '<input type="checkbox" style=" vertical-align: middle; margin: 0 0.2em 0.25em -1.6em; font-size: 16px; " disabled> ')
                    .replace(/^\s*\[x\]\s*/, '<input type="checkbox" style=" vertical-align: middle; margin: 0 0.2em 0.25em -1.6em; font-size: 16px; " checked disabled> ');
                return '<li style="list-style: none">' + text + "</li>";
            }
            else {
                return "<li>" + text + "</li>";
            }
        };
    };
    MarkdownService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    MarkdownService.ctorParameters = function () { return [
        { type: HttpClient, },
    ]; };
    return MarkdownService;
}());
export { MarkdownService };
