// ngx-md/markdown/markdown.module.js
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { HttpClientModule } from "@angular/common/http";
import { MarkdownComponent } from "./markdown.component";
import { MarkdownService } from "./markdown.service";
import { MarkdownConfig } from "./markdown.config";
var MarkdownModule = (function () {
    function MarkdownModule() {
    }
    MarkdownModule.forRoot = function () {
        return {
            ngModule: MarkdownModule,
            providers: [MarkdownConfig]
        };
    };
    MarkdownModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule, HttpClientModule],
                    declarations: [MarkdownComponent],
                    providers: [MarkdownService],
                    exports: [CommonModule, MarkdownComponent, HttpClientModule],
                    entryComponents: [MarkdownComponent]
                },] },
    ];
    /** @nocollapse */
    MarkdownModule.ctorParameters = function () { return []; };
    return MarkdownModule;
}());
export { MarkdownModule };
